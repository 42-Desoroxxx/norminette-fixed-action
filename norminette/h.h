#define stuff(x) x = { \
0, 2, 3, 4, 5 }
