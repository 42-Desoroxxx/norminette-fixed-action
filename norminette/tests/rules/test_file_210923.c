int	main(void)
{
	int		a;
	int		b;

	(a = 4) && (b = 6);
}

#include <stdio.h>

int	main(void)
{
	int	a;

	(a = 0, printf("%d\n", a));
	return (0);
}

int	v(int *restrict t)
{
	const char	*restrict s = "Hello World";

	printf("%s\n", s), a = 15;
	return (0);
}

int	main(void)
{
	int	a;

	a = 0, printf("%d\n", a);
	return (0);
}
