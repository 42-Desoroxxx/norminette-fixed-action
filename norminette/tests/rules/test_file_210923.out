Missing or invalid header. Header are being reintroduced as a mandatory part of your files. This is not yet an error.
[36mtest_file_210923.c[0m - [32mIsFuncDeclaration[0m In "GlobalScope" from "None" line 1":
		<INT> <TAB> <IDENTIFIER=main> <LPARENTHESIS> <VOID> <RPARENTHESIS> <NEWLINE>
[36mtest_file_210923.c[0m - [32mIsBlockStart[0m In "Function" from "GlobalScope" line 2":
		<LBRACE> <NEWLINE>
[36mtest_file_210923.c[0m - [32mIsVarDeclaration[0m In "Function" from "GlobalScope" line 3":
		<TAB> <INT> <TAB> <TAB> <IDENTIFIER=a> <SEMI_COLON> <NEWLINE>
[36mtest_file_210923.c[0m - [32mIsVarDeclaration[0m In "Function" from "GlobalScope" line 4":
		<TAB> <INT> <TAB> <TAB> <IDENTIFIER=b> <SEMI_COLON> <NEWLINE>
[36mtest_file_210923.c[0m - [32mIsEmptyLine[0m In "Function" from "GlobalScope" line 5":
		<NEWLINE>
[36mtest_file_210923.c[0m - [32mIsAssignation[0m In "Function" from "GlobalScope" line 6":
		<TAB> <LPARENTHESIS> <IDENTIFIER=a> <SPACE> <ASSIGN> <SPACE> <CONSTANT=4> <RPARENTHESIS> <SPACE> <AND> <SPACE> 
[36mtest_file_210923.c[0m - [32mIsAssignation[0m In "Function" from "GlobalScope" line 6":
		<LPARENTHESIS> <IDENTIFIER=b> <SPACE> <ASSIGN> <SPACE> <CONSTANT=6> <RPARENTHESIS> <SEMI_COLON> <NEWLINE>
[36mtest_file_210923.c[0m - [32mIsBlockEnd[0m In "Function" from "GlobalScope" line 7":
		<RBRACE> <NEWLINE>
[36mtest_file_210923.c[0m - [32mIsEmptyLine[0m In "GlobalScope" from "None" line 8":
		<NEWLINE>
[36mtest_file_210923.c[0m - [32mIsPreprocessorStatement[0m In "GlobalScope" from "None" line 9":
		<INCLUDE=#include <stdio.h>> <NEWLINE>
[36mtest_file_210923.c[0m - [32mIsEmptyLine[0m In "GlobalScope" from "None" line 10":
		<NEWLINE>
[36mtest_file_210923.c[0m - [32mIsFuncDeclaration[0m In "GlobalScope" from "None" line 11":
		<INT> <TAB> <IDENTIFIER=main> <LPARENTHESIS> <VOID> <RPARENTHESIS> <NEWLINE>
[36mtest_file_210923.c[0m - [32mIsBlockStart[0m In "Function" from "GlobalScope" line 12":
		<LBRACE> <NEWLINE>
[36mtest_file_210923.c[0m - [32mIsVarDeclaration[0m In "Function" from "GlobalScope" line 13":
		<TAB> <INT> <TAB> <IDENTIFIER=a> <SEMI_COLON> <NEWLINE>
[36mtest_file_210923.c[0m - [32mIsEmptyLine[0m In "Function" from "GlobalScope" line 14":
		<NEWLINE>
[36mtest_file_210923.c[0m - [32mIsAssignation[0m In "Function" from "GlobalScope" line 15":
		<TAB> <LPARENTHESIS> <IDENTIFIER=a> <SPACE> <ASSIGN> <SPACE> <CONSTANT=0> <COMMA> <SPACE> 
[36mtest_file_210923.c[0m - [32mIsFunctionCall[0m In "Function" from "GlobalScope" line 15":
		<IDENTIFIER=printf> <LPARENTHESIS> <STRING="%d\n"> <COMMA> <SPACE> <IDENTIFIER=a> <RPARENTHESIS> <RPARENTHESIS> <SEMI_COLON> <NEWLINE>
[36mtest_file_210923.c[0m - [32mIsExpressionStatement[0m In "Function" from "GlobalScope" line 16":
		<TAB> <RETURN> <SPACE> <LPARENTHESIS> <CONSTANT=0> <RPARENTHESIS> <SEMI_COLON> <NEWLINE>
[36mtest_file_210923.c[0m - [32mIsBlockEnd[0m In "Function" from "GlobalScope" line 17":
		<RBRACE> <NEWLINE>
[36mtest_file_210923.c[0m - [32mIsEmptyLine[0m In "GlobalScope" from "None" line 18":
		<NEWLINE>
[36mtest_file_210923.c[0m - [32mIsFuncDeclaration[0m In "GlobalScope" from "None" line 19":
		<INT> <TAB> <IDENTIFIER=v> <LPARENTHESIS> <INT> <SPACE> <MULT> <RESTRICT> <SPACE> <IDENTIFIER=t> <RPARENTHESIS> <NEWLINE>
[36mtest_file_210923.c[0m - [32mIsBlockStart[0m In "Function" from "GlobalScope" line 20":
		<LBRACE> <NEWLINE>
[36mtest_file_210923.c[0m - [32mIsVarDeclaration[0m In "Function" from "GlobalScope" line 21":
		<TAB> <CONST> <SPACE> <CHAR> <TAB> <MULT> <RESTRICT> <SPACE> <IDENTIFIER=s> <SPACE> <ASSIGN> <SPACE> <STRING="Hello World"> <SEMI_COLON> <NEWLINE>
[36mtest_file_210923.c[0m - [32mIsEmptyLine[0m In "Function" from "GlobalScope" line 22":
		<NEWLINE>
[36mtest_file_210923.c[0m - [32mIsFunctionCall[0m In "Function" from "GlobalScope" line 23":
		<TAB> <IDENTIFIER=printf> <LPARENTHESIS> <STRING="%s\n"> <COMMA> <SPACE> <IDENTIFIER=s> <RPARENTHESIS> <COMMA> <SPACE> 
[36mtest_file_210923.c[0m - [32mIsAssignation[0m In "Function" from "GlobalScope" line 23":
		<IDENTIFIER=a> <SPACE> <ASSIGN> <SPACE> <CONSTANT=15> <SEMI_COLON> <NEWLINE>
[36mtest_file_210923.c[0m - [32mIsExpressionStatement[0m In "Function" from "GlobalScope" line 24":
		<TAB> <RETURN> <SPACE> <LPARENTHESIS> <CONSTANT=0> <RPARENTHESIS> <SEMI_COLON> <NEWLINE>
[36mtest_file_210923.c[0m - [32mIsBlockEnd[0m In "Function" from "GlobalScope" line 25":
		<RBRACE> <NEWLINE>
[36mtest_file_210923.c[0m - [32mIsEmptyLine[0m In "GlobalScope" from "None" line 26":
		<NEWLINE>
[36mtest_file_210923.c[0m - [32mIsFuncDeclaration[0m In "GlobalScope" from "None" line 27":
		<INT> <TAB> <IDENTIFIER=main> <LPARENTHESIS> <VOID> <RPARENTHESIS> <NEWLINE>
[36mtest_file_210923.c[0m - [32mIsBlockStart[0m In "Function" from "GlobalScope" line 28":
		<LBRACE> <NEWLINE>
[36mtest_file_210923.c[0m - [32mIsVarDeclaration[0m In "Function" from "GlobalScope" line 29":
		<TAB> <INT> <TAB> <IDENTIFIER=a> <SEMI_COLON> <NEWLINE>
[36mtest_file_210923.c[0m - [32mIsEmptyLine[0m In "Function" from "GlobalScope" line 30":
		<NEWLINE>
[36mtest_file_210923.c[0m - [32mIsAssignation[0m In "Function" from "GlobalScope" line 31":
		<TAB> <IDENTIFIER=a> <SPACE> <ASSIGN> <SPACE> <CONSTANT=0> <COMMA> <SPACE> 
[36mtest_file_210923.c[0m - [32mIsFunctionCall[0m In "Function" from "GlobalScope" line 31":
		<IDENTIFIER=printf> <LPARENTHESIS> <STRING="%d\n"> <COMMA> <SPACE> <IDENTIFIER=a> <RPARENTHESIS> <SEMI_COLON> <NEWLINE>
[36mtest_file_210923.c[0m - [32mIsExpressionStatement[0m In "Function" from "GlobalScope" line 32":
		<TAB> <RETURN> <SPACE> <LPARENTHESIS> <CONSTANT=0> <RPARENTHESIS> <SEMI_COLON> <NEWLINE>
[36mtest_file_210923.c[0m - [32mIsBlockEnd[0m In "Function" from "GlobalScope" line 33":
		<RBRACE> <NEWLINE>
test_file_210923.c: Error!
Error: MULT_ASSIGN_LINE     (line:   6, col:  16):	Multiple assignations on a single line
Error: TOO_FEW_TAB          (line:   6, col:  16):	Missing tabs for indent level
Error: TOO_MANY_INSTR       (line:   6, col:  16):	Too many instructions on a single line
Error: INCLUDE_START_FILE   (line:   9, col:   1):	Include must be at the start of file
Error: TOO_FEW_TAB          (line:  15, col:  13):	Missing tabs for indent level
Error: TOO_MANY_INSTR       (line:  15, col:  13):	Too many instructions on a single line
Error: TOO_FEW_TAB          (line:  23, col:  23):	Missing tabs for indent level
Error: TOO_MANY_INSTR       (line:  23, col:  23):	Too many instructions on a single line
Error: TOO_FEW_TAB          (line:  31, col:  12):	Missing tabs for indent level
Error: TOO_MANY_INSTR       (line:  31, col:  12):	Too many instructions on a single line
